### ISSUE TITLE
Keyboard stutters

### ISSUE THREAD

**Issue by AverageGuy** [debauchee/barrier#269]
### Operating Systems ###

Server: Mint Mate 18
Client: Mint Mate 19

### Barrier Version ###

1.9.0

### Steps to reproduce bug ###
Start typing, sooner or latter you will type a letter, number, punctuation or space and magically 2 or more of them will appear.  It's fairly intermittent.  I typed only one t in "later", for instance.  

**READ ME, DELETE ME**: Try to be succinct. If your bug is intermittent, try and describe what you're doing when it happens most.

### Other info ###

* When did the problem start to occur? After I installeddd barrier on both systems from source.
* Is there a way to work around it? No, other than using the backspace key.
* Does this bug prevent you from using Barrier entirely? No
Put anything else you can think of here.
It affects anything you aree typing into.  This bug, a terminal window, Vim instance, are exampllles of where it happens, and no I'm not holding the key down.  I'm a touch typiist and get on/off the keys pretty fast.

**Comment by AverageGuy** [comment]
I don''''t undeeeerstand why I'm here

**Comment by AverageGuy** [comment]
I have no idea why it happeeened.  I got something that didn''''t belong to me in the first attempt at describing the bug.

**Comment by AverageGuy** [comment]
All works OK on the server.  No stuttering.  I replaced synergy with barrieer but never saw it with synerrgy.  The client is a new machine running Mate 19 annd I was unable to get synergy to run on it.  Incompatible versions it said.  That's why I am runniiing barrier....
I didn't type but onnnnne period.

**Comment by AdrianKoshka** [comment]
Are you trying to use barrier over wifi?

**Comment by AverageGuy** [comment]
Both systems are connected to a gigabit swiiiitch via ethernet.

**Comment by AverageGuy** [comment]
At a different location, I have the same issue.  Mint 19 on one machine, Mint 18 on the other.  The only difference is the master is running 19 at this location.  The original problem exists when the master is on 18.  Stuttering is on the slave in both cases.

**Comment by charliehorse55** [comment]
I also have this problem. I have a VFIO setup, and I use barrier to share the mouse between OSes. The server runs on linux, and the guest is either windows 10 or another linux instance. The stuttering is ONLY present on the linux guest, windows works fine. In both cases, the machines are connected through a virtualized 10gbit link, so I highly doubt this is a networking issue.

**Comment by charliehorse55** [comment]
Does anyone have any information on this? The stutters can be quite long for me (several seconds). If I happened to be pressing a key while editing code, it will mess up what I'm editing. This happens multiple times per hour. Luckily I can ctrl-z, but it breaks my flow. 

It's really funny when it gets stuck on backspace and deletes 10+ lines of code :)

**Comment by amerlyq** [comment]
TL;DR: use `xset r off` on client.

BET:DEV: fix barrier(1) client code to disable autorepeat on XTest simulated keys (only simulated by it itself).

I use latest version barrier with `ArchLinux PC -> ArchLinux Laptop` over home router 5G wifi.
I strongly suspect the root problem lays in "X11 autorepeat" feature working *both* on server and client.
Barrier sends keydown and keyup events separately, and if network latency becomes more than autorepeat trigger on client -- it will result in wrong autorepeat. It's easy to verify: 
* use `xset r rate 80 20` on client to observe the issue on almost each keypress
* use `xset r off` -- and issue on client is gone.

There is no need to disable autorepeat on server -- its events will propagate to client and "simulate" autorepeat there.

**Comment by charliehorse55** [comment]
Unfortunately, that's not the problem. The mouse always freezes along with the keyboard input.

**Comment by p12tic** [comment]
Looks like a duplicate of #617 which is fixed in #679.

**Comment by deters** [comment]
Probably wifi lag... barrier could reduce this by implementing a "keep_alive" option: sending dummy packets at a constant rate to the client in order to keep wifi module active. 

My current workaround:

starting netcat in the server: 
nc -l 5555

using netcat in the client to send a "." to the server every 50ms:
while [ 1 ]; do sleep 0.05; echo .; done | nc 192.168.1.190 5555

**Comment by charliehorse55** [comment]
This got fixed at some point by a PR i remember reading
